src/ragged/_spec_set_functions.py

```diff
@@ -47,8 +52,31 @@ def unique_all(x: array, /) -> tuple[array, array, array, array]:
     https://data-apis.org/array-api/latest/API_specification/generated/array_api.unique_all.html
     """
 
-    x  # noqa: B018, pylint: disable=W0104
-    raise NotImplementedError("TODO 128")  # noqa: EM101
+    if isinstance(x, ragged.array):
+        if len(x) == 1:
+            return unique_all_result(
+                values=ragged.array(x),
+                indices=ragged.array([0]),
+                inverse_indices=ragged.array([0]),
+                counts=ragged.array([1]),
+            )
+        else:
+            x_flat = ak.ravel(x._impl)
+            values, indices, inverse_indices, counts = np.unique(
+                x_flat.layout.data,
+                return_index=True,
+                return_inverse=True,
+                return_counts=True,
+            )
```

Comment:
I'm pretty sure that this `np.unique` call also needs an `equal_nan=False` to be equivalent to [np.unique_all](https://numpy.org/doc/stable/reference/generated/numpy.unique_all.html). Same for the others.